app/src/main/java/com/khaithu/a3ctech_android_final_project/presenter/GetMoviePresenter.kt

```diff
@@ -0,0 +1,55 @@
+package com.khaithu.a3ctech_android_final_project.presenter
+
+import android.view.View
+import com.khaithu.a3ctech_android_final_project.api.RestClient
+import com.khaithu.a3ctech_android_final_project.model.PageMovie
+import com.khaithu.a3ctech_android_final_project.model.ResultMovie
+import com.khaithu.a3ctech_android_final_project.model.UrlApi
+import com.khaithu.a3ctech_android_final_project.model.enum.LanguageEnum
+import com.khaithu.a3ctech_android_final_project.presenter.interfacePre.IGetMoviePresenter
+import com.khaithu.a3ctech_android_final_project.service.IMovieService
+import com.khaithu.a3ctech_android_final_project.view.interfaceView.IHomePageFragment
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+
+class GetMoviePresenter(var iHomePageFragment: IHomePageFragment) : IGetMoviePresenter {
+
+    lateinit var retrofit: Retrofit
+    lateinit var movieService: IMovieService
+    lateinit var litOfMovies: MutableList<ResultMovie>
+
+    var urlApi: UrlApi = UrlApi(
+        "https://api.themoviedb.org/",
+        2,
+        "0e47cd8143d6b9a833d0258baff7d491",
+        LanguageEnum.ENGLISH,
+        "popular"
+    )
+
+    override fun getMovie(view: View) {
+        retrofit = RestClient.getInstanceRetrofit(urlApi.baseUrl)!!
+        movieService = retrofit.create(IMovieService::class.java)
+
+        var call: Call<PageMovie> = movieService.listOfMovies(
+            urlApi.category,
+            urlApi.apiKey,
+            urlApi.language.getValue(),
+            1
+        )
+        call.enqueue(object : Callback<PageMovie> {
+            override fun onResponse(call: Call<PageMovie>, response: Response<PageMovie>) {
+                var pageMovie: PageMovie? = response.body()
+                if (pageMovie != null) {
+                    litOfMovies = pageMovie.result
+                    iHomePageFragment.onMoviesReceived(litOfMovies, view)
+                }
+            }
+
+            override fun onFailure(call: Call<PageMovie>, t: Throwable) {
+                println()
+            }
```

Comment:
You should define method `onError()` at `View`
```suggestion
            override fun onFailure(call: Call<PageMovie>, t: Throwable) {
                view?.onError(< message or anything you want>)
            }
```